### PR TITLE
DEV: switch to using subscription gem to avoid the need to rely on Subscription Client plugin

### PR DESCRIPTION
**Pull request by merefield**
* IMPROVE: moves CW to rely on referenced gem instead of Subscription Client, so it can run entirely independently.
* IMPROVE: rationalises and consolidates subscription logic somewhat:
  * COMPATIBILITY: moves FE subscriptions management to a Service (in part because Mixins are deprecated)
  * IMPROVE: moves subscription serialisation to a dedicated controller on the BE.
* COMPATIBILITY: converts several related Components to Glimmer (and colocates template files as per standard).
* alters the way subscription client is mocked up in tests to accommodate the fact that overriding the class does not work the same way now that we've added the gem
* adds FE tests for Authorization and Subscription status.
* COMPATIBILITY: fixes deprecation of registerUnbound with Wizard Char Counter
* FIX: non working FE tests because of refactor of Core Composer has changed CSS selectors.
* IMPROVE: moves Pavilion icon to standard SVG registration system (from ad hoc template addition).

### CODE REVIEW

**Comment by merefield** [comment]
@angusmcleod remaining failing tests are not related to this PR and are also failing in Production.  I've not tackled them because they are out of scope.

**Comment by angusmcleod** [comment]
@merefield Nice. Ping me when it's ready for review.

**Comment by merefield** [comment]
@angusmcleod, these front end tests are not failing locally on two types of browser, any ideas?

**Comment by angusmcleod** [comment]
Not sure.

@jumagura Any idea why these tests are failing?

**Comment by merefield** [comment]
@jumagura please ignore, I found the problem, it's a breaking change from core due to updates to Composer and Modal selectors.

@angusmcleod this is now ready for review and merge.

**Review by angusmcleod — CHANGES_REQUESTED** [review]
@merefield Two minor changes and a comment for future reference (no need to convert to .gjs in this PR).

File: assets/javascripts/discourse/components/wizard-subscription-badge.hbs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<DButton
+  @action={{this.click}}
+  class="wizard-subscription-badge {{this.subscription.subscriptionType}}"
+  @title={{this.title}}
+  {{did-insert this.update}}
+>
+  <svg
```
[the remark this answers is not in the document]
Comment (merefield) [reply assets/javascripts/discourse/components/wizard-subscription-badge.hbs:7]: can do ... I just re-used what was there already.  I'll address.

[the remark this answers is not in the document]
Comment (merefield) [reply assets/javascripts/discourse/components/wizard-subscription-badge.hbs]: this is both resolved and improved

File: assets/javascripts/discourse/components/wizard-subscription-cta.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { inject as service } from "@ember/service";
```
[the remark this answers is not in the document]
Comment (merefield) [reply assets/javascripts/discourse/components/wizard-subscription-cta.js:1]: Glimmer split was all the rage only a few weeks ago ... can't keep up!

File: lib/custom_wizard/subscription.rb [hunk 3]
```diff
@@ -176,10 +180,6 @@ def community?
     product_slug === "community"
   end
 
-  def self.client_installed?
```
[the remark this answers is not in the document]
Comment (merefield) [reply lib/custom_wizard/subscription.rb:179]: this is restored ... please check it's enough @angusmcleod

**Comment by merefield** [comment]
all requested changes implemented @angusmcleod ...

**Comment by angusmcleod** [comment]
@merefield Looks like we've got a failing check here.

**Comment by merefield** [comment]
resolved @angusmcleod

**Review by angusmcleod — APPROVED** [review]

**Pull request merged**